klog_test.go

```diff
@@ -820,6 +820,60 @@ func TestInfoS(t *testing.T) {
 	}
 }
 
+// Test that Verbose.InfoS works as advertised.
+func TestVInfoS(t *testing.T) {
+	setFlags()
+	defer logging.swap(logging.newBuffers())
+	timeNow = func() time.Time {
+		return time.Date(2006, 1, 2, 15, 4, 5, .067890e9, time.Local)
+	}
+	pid = 1234
+	var testDataInfo = []struct {
+		msg        string
+		format     string
+		keysValues []interface{}
+	}{
+		{
+			msg:        "test",
+			format:     "I0102 15:04:05.067890    1234 klog_test.go:%d] \"test\" pod=\"kubedns\"\n",
+			keysValues: []interface{}{"pod", "kubedns"},
+		},
+		{
+			msg:        "test",
+			format:     "I0102 15:04:05.067890    1234 klog_test.go:%d] \"test\" replicaNum=20\n",
+			keysValues: []interface{}{"replicaNum", 20},
+		},
+		{
+			msg:        "test",
+			format:     "I0102 15:04:05.067890    1234 klog_test.go:%d] \"test\" err=\"test error\"\n",
+			keysValues: []interface{}{"err", errors.New("test error")},
+		},
+	}
+
+	logging.verbosity.Set("2")
+	defer logging.verbosity.Set("0")
+
+	for l := Level(0); l < Level(4); l++ {
+		for _, data := range testDataInfo {
+			logging.file[infoLog] = &flushBuffer{}
+
+			V(l).InfoS(data.msg, data.keysValues...)
+			_, _, line, _ := runtime.Caller(0) //Fragile code, don't move away form the line above
```

Comment:
Hmm have you considered using `gmtSscanf` to avoid using `runtime.Caller` ? Look at test below `TestErrorS` it doesn't use it.

Reply:
I did not find the `scanf` very clear and it gave an odd error when failing to parse the string. But I'm happy to match the current test cases. Consistency is best. The `scanf` has the advantage that it's not fragile in the same way as `runtime.Caller`.

Reply:
I've updated the test to better match the existing tests. It does feel a bit wrong to to build the `want` using data from `got`, but it's just the line number and strictly controlled so probably not that bad.